aws-frauddetector-detector/src/aws_frauddetector_detector/helpers/model_helpers.py

```diff
@@ -535,13 +535,39 @@ def _get_entity_types_and_return_entity_types_model(
 def get_external_model_arns_from_model(model: models.ResourceModel) -> Set[str]:
     if model.AssociatedModels is None:
         return set()
-    return {m.Arn for m in model.AssociatedModels if util.is_arn_external_model_arn(m.Arn)}
+    return {m.Arn for m in model.AssociatedModels if util.is_external_model_arn(m.Arn)}
 
 
 def get_external_model_endpoints_from_model(model: models.ResourceModel) -> List[str]:
     if model.AssociatedModels is None:
         return []
-    return [util.extract_name_from_arn(m.Arn) for m in model.AssociatedModels if util.is_arn_external_model_arn(m.Arn)]
+    return [util.extract_name_from_arn(m.Arn) for m in model.AssociatedModels if util.is_external_model_arn(m.Arn)]
+
+
+# Model Versions for Detector
+
+
+def get_model_versions_from_model(model: models.ResourceModel) -> List[dict]:
+    if model.AssociatedModels is None:
+        return []
+
+    model_versions = []
+
+    for m in model.AssociatedModels:
+        model_id, model_type, model_version_number = get_model_version_details_from_arn(m.Arn)
+        model_versions.append(
+            {"modelId": model_id, "modelType": model_type, "modelVersionNumber": model_version_number}
+        )
+
+    return model_versions
+
+
+def get_model_version_details_from_arn(arn):
+    segmented_arn = arn.split("/")
+    model_version_number = segmented_arn[-1]
+    model_id = segmented_arn[-2]
+    model_type = segmented_arn[-3]
+    return model_id, model_type, model_version_number
```

Comment:
nit: I think this should go into `util`, as it doesn't involve a model, and it is similar to the other methods in `util` that deal with Arns.

Also, should we double check arn split length or double check `is_model_version_arn` inside of this method? I'm just thinking that we will probably never want to use this method unless we know the arn is a valid model version arn, since it has the potential to throw index error.

Reply:
Relocating to util, and combining with `is_model_version_arn `

Reply:
As discussed offline, I could be persuaded to not have the redundant check, but I like the redundancy here, and there is no real cost for also having a `util` method throw an exception. I'd rather have the method throw an exception than have the possibility to throw an index error that would bubble up to the customer in a very cryptic (non-helpful) way.

Reply:
- Relocated `get_model_version_details_from_arn` to util, as it works with ARNs.

- Before split(), check added for InvalidRequest to ensure ARN at least contains 4 segments, and `model-version/`, to mitigate an index error.